File: app/src/main/java/com/example/floating/bakingapp/database/RecipeDbHelper.java
package com.example.floating.bakingapp.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.example.floating.bakingapp.data.Recipe;
import com.example.floating.bakingapp.utils.RecipeUtils;

import static com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.COLUMN_INGREDIENT_LIST;
import static com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.COLUMN_RECIPE_NAME;
import static com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.TABLE_NAME;

/**
 * Copyright (c) Abdulkarim Abdulrahman Ayoola on 6/14/2017.
 */
public  class RecipeDbHelper extends SQLiteOpenHelper {

    //Database version
    private static final int VERSION = 1;

    //Database name
    private static final String DATABASE_NAME = "recipe.db";

    public RecipeDbHelper(Context context) {
        super(context, DATABASE_NAME, null, VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {

        //Create Table SQLite Query string
        final String CREATE_TABLE_RECIPE = "CREATE TABLE "
                + TABLE_NAME + "("
                + RecipeContract.RecipeEntry._ID + " INTEGER PRIMARY KEY,"
                + COLUMN_RECIPE_NAME + " TEXT NOT NULL,"
                + COLUMN_INGREDIENT_LIST + " TEXT NOT NULL"
                + ");";
        sqLiteDatabase.execSQL(CREATE_TABLE_RECIPE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int oldVersion, int newVersion) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
        onCreate(sqLiteDatabase);
    }

    //Table methods for widget purposes

    public long addViewedRecipe(Recipe recipe) {

        SQLiteDatabase db = this.getWritableDatabase();

        ContentValues values = new ContentValues();

        values.put(COLUMN_RECIPE_NAME, recipe.getName());
        values.put(COLUMN_INGREDIENT_LIST, RecipeUtils.getIngredientsString(recipe.getIngredients()));

        return db.insert(TABLE_NAME, null, values);
    }

    public long updateRecipe(Recipe recipe) {

        SQLiteDatabase db = this.getWritableDatabase();

        ContentValues values = new ContentValues();
        values.put(COLUMN_RECIPE_NAME, recipe.getName());
        values.put(COLUMN_INGREDIENT_LIST, RecipeUtils.getIngredientsString(recipe.getIngredients()));

        String selection = RecipeContract.RecipeEntry._ID + "=?";
        String[] selectrionArgs = {String.valueOf(0)};

        long updateId = db.update(
                TABLE_NAME,
                values,
                selection,
                selectrionArgs
        );

        db.close();
        return updateId;
    }

    public Recipe getViewedRecipe() {
            SQLiteDatabase db = this.getReadableDatabase();

            Cursor cursor = db.query(
                    TABLE_NAME,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null
            );

            Recipe recipe = new Recipe();

            if (cursor.moveToFirst()) {
                do {
                    int name_index = cursor.getColumnIndex(COLUMN_RECIPE_NAME);
                    int ingredient_index = cursor.getColumnIndex(COLUMN_INGREDIENT_LIST);
                    recipe.setName(cursor.getString(name_index));
                    recipe.setIngredient_string(cursor.getString(ingredient_index));
                } while (cursor.moveToNext());
            }
            cursor.close();
            return recipe;
        }

    public int getRecipeDBSize(int id) {
            SQLiteDatabase db = this.getReadableDatabase();

            Cursor cursor = db.query(
                    TABLE_NAME,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null
            );

            int size = cursor.getCount();
            cursor.close();

            return size;
        }
}

File: app/src/main/java/com/example/floating/bakingapp/database/RecipeDbHelper.kt
package com.example.floating.bakingapp.database

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

import com.example.floating.bakingapp.model.Recipe
import com.example.floating.bakingapp.utils.RecipeUtils

import com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.Companion.COLUMN_INGREDIENT_LIST
import com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.Companion.COLUMN_RECIPE_NAME
import com.example.floating.bakingapp.database.RecipeContract.RecipeEntry.Companion.TABLE_NAME
import com.example.floating.bakingapp.model.Ingredients
import com.example.floating.bakingapp.model.Steps

/**
 * Copyright (c) Abdulkarim Abdulrahman Ayoola on 6/14/2017.
 */
class RecipeDbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, VERSION) {

    val viewedRecipe: Recipe
        get() {
            val db = this.readableDatabase

            val cursor = db.query(
                    TABLE_NAME,
                    null, null, null, null, null, null
            )

            val recipe = Recipe(0, "", null, null, 0, "", "")

            if (cursor.moveToFirst()) {
                do {
                    val nameIndex = cursor.getColumnIndex(COLUMN_RECIPE_NAME)
                    val ingredientIndex = cursor.getColumnIndex(COLUMN_INGREDIENT_LIST)
                    recipe.name = cursor.getString(nameIndex)
                    recipe.ingredient_string = cursor.getString(ingredientIndex)
                } while (cursor.moveToNext())
            }
            cursor.close()
            return recipe
        }

    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {

        //Create Table SQLite Query string
        val CREATE_TABLE_RECIPE = ("CREATE TABLE "
                + TABLE_NAME + "("
                + RecipeContract.RecipeEntry._ID + " INTEGER PRIMARY KEY,"
                + COLUMN_RECIPE_NAME + " TEXT NOT NULL,"
                + COLUMN_INGREDIENT_LIST + " TEXT NOT NULL"
                + ");")
        sqLiteDatabase.execSQL(CREATE_TABLE_RECIPE)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME)
        onCreate(sqLiteDatabase)
    }

    //Table methods for widget purposes

    fun addViewedRecipe(recipe: Recipe): Long {

        val db = this.writableDatabase

        val values = ContentValues()

        values.put(COLUMN_RECIPE_NAME, recipe.name)
        values.put(COLUMN_INGREDIENT_LIST, RecipeUtils.getIngredientsString(recipe.ingredients))

        return db.insert(TABLE_NAME, null, values)
    }

    fun updateRecipe(recipe: Recipe): Long {

        val db = this.writableDatabase

        val values = ContentValues()
        values.put(COLUMN_RECIPE_NAME, recipe.name)
        values.put(COLUMN_INGREDIENT_LIST, RecipeUtils.getIngredientsString(recipe.ingredients))

        val selection = RecipeContract.RecipeEntry._ID + "=?"
        val selectionArgs = arrayOf(0.toString())

        val updateId = db.update(
                TABLE_NAME,
                values,
                selection,
                selectionArgs
        ).toLong()

        db.close()
        return updateId
    }

    fun getRecipeDBSize(id: Int): Int {
        val db = this.readableDatabase

        val cursor = db.query(
                TABLE_NAME, null, null, null, null,
                null, null)

        val size = cursor.count
        cursor.close()

        return size
    }

    companion object {

        //Database version
        private const val VERSION = 1

        //Database name
        private const val DATABASE_NAME = "recipe.db"
    }
}
